docs/source/configuration/authorization.mdx

```diff
@@ -408,6 +416,192 @@ The response retains the initial request's shape but returns `null` for unauthor
+
+```graphql
+extend schema
+  @link(
+    url: "https://specs.apollo.dev/federation/v2.5",
+    import: [..., "@policy"])
+```
+
+The `@policy` directive is defined as follows:
+
+```graphql
+scalar federation__Policy
+directive @policy(policies: [[federation__Policy!]!]!) on OBJECT | FIELD_DEFINITION | INTERFACE | SCALAR | ENUM
+```
+
+Using the `@policy` directive requires a [Supergraph plugin](../customizations/overview) to evaluate the authorization policies. You can do this with a [Rhai script](../customizations/rhai/) or [coprocessor](../customizations/coprocessor). Refer to the following [Rhai script](#usage-with-a-rhai-script) and [coprocessor](#usage-with-a-coprocessor) examples for more information. (Although a [native plugin](../customizations/native) can also evaluate authorization policies, we don't recommend using it.)
+
+#### Combining policies with `AND`/`OR` logic
+
+Authorization validation uses **AND** logic between the elements in the inner-level `policies` array, where a request must include _all_ elements in the inner-level `policies` array to resolve the associated field or type. For the following example, a request would need `policy1` **AND** `policy2` **AND** `policy3` to be authorized:
+
+```graphql
+@policy(policies: [["policy1", "policy2", "policy3"]])
+```
+
+Alternatively, to introduce **OR** logic you can use nested arrays. For the following example, a request would need `policy1` **OR** `policy2` **OR** `policy3` to be authorized:
+
+```graphql
+@policy(policies: [["policy1"], ["policy2"], ["policy3"]])
+```
+
+You can nest arrays and elements as needed to achieve your desired logic. For the following example, its syntax requires requests to have either (`policy1` **AND** `policy2`) **OR** just `policy3` to be authorized:
+
+```graphql
+@policy(policies: [["policy", "policy2"], ["policy3"]])
+```
+
+##### Usage with a Rhai script
+
+The `policies` argument contains a list of strings with no formatting constraints, meaning you can use them to store Rhai code.
+
+<ExpansionPanel title="Click to expand">
+
+As an example, the following schema defines policies as boolean expressions that can be evaluated in Rhai:
+
+```graphql
+type Query {
+  me: User @join__field(graph: ACCOUNTS) @policy(policies: [["claims[`kind`] == `user`"]])
+}
+
+type User
+  @join__owner(graph: ACCOUNTS)
+  @join__type(graph: ACCOUNTS, key: "id") {
+  id: ID! @join__field(graph: ACCOUNTS)
+  username: String @join__field(graph: ACCOUNTS)
+  username: String @join__field(graph: ACCOUNTS) @policy(policies: [["claims[`roles`].contains(`support`)"]])
+}
+```
+
+You can then use the following Rhai script to evaluate the policies:
+
+```
+fn supergraph_service(service) {
+  let request_callback = |request| {
+    let claims = request.context["apollo_authentication::JWT::claims"];
+    let policies = request.context["apollo_authorization::policies::required"];
+    for key in policies.keys() {
```

Comment:
`policies.keys()` can be empty which results in a type error that breaks execution for any operations without policies. We need to adjust this to handle that case (as part of rewriting the `eval` bits)

Reply:
fixed in https://github.com/apollographql/router/pull/4256